---

- name: Create and address loopback interface
  hosts: iosxe
  gather_facts: no

  tasks:
  - name: Create loopback 22
    cisco.ios.ios_interfaces:
      config:
      - name: Loopback{{item}}
        description: Created by ansible
        enabled: true
    
    loop:
        - 11
        - 12
        - 13  
  
  - name: Assign IP to loopback interface
    cisco.ios.ios_l3_interfaces:
      config:
      - name: Loopback{{item}}
        ipv4:
        - address: 192.10.{{item}}.{{item}}/24
      
    loop:
        - 11
        - 12
        - 13
    register: subip

  - name: Save running to startup when modified
    cisco.ios.ios_config:
      save_when: modified

##
## Line - address: 192.10.{{item}}.{{item}}/24 was causing problem with subnet_id. 
## Looked like value of {{item}} was creating a subnet conflict with next interface.
## However, after running same playbook a second time and subsequent times playbook worked fine.
## 
